### PR TITLE
8303229: JFR: Preserve disk repository after exit

### PR DESCRIPTION
**Pull request by egahlin**
Hi,

Could I have a review of an enhancement that adds the option preserve-repository=true/false to -XX:FlightRecorderOptions and jcmd JFR.configure. When set to true, chunk files in the repository will not be removed by the JVM at exit.

Testing: jdk/jdk/jfr

Thanks
Erik

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8303754](https://bugs.openjdk.org/browse/JDK-8303754) to be approved

### Issues
 * [JDK-8303229](https://bugs.openjdk.org/browse/JDK-8303229): JFR: Preserve disk repository after exit
 * [JDK-8303754](https://bugs.openjdk.org/browse/JDK-8303754): JFR: Preserve disk repository after exit (**CSR**)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Markus Grönlund](https://openjdk.org/census#mgronlun) (@mgronlun - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13111/head:pull/13111` \
`$ git checkout pull/13111`

Update a local copy of the PR: \
`$ git checkout pull/13111` \
`$ git pull https://git.openjdk.org/jdk.git pull/13111/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13111`

View PR using the GUI difftool: \
`$ git pr show -t 13111`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13111.diff">https://git.openjdk.org/jdk/pull/13111.diff</a>

</details>

### CODE REVIEW

**Comment by dholmes-ora** [comment]
@egahlin you to update this with latest master to get DCmd `num_arguments` changes that went in recently.

**Comment by dholmes-ora** [comment]
Also I would expect to see an update to the JFR manpage for this. Thanks.

**Comment by egahlin** [comment]
> @egahlin you to update this with latest master to get DCmd `num_arguments` changes that went in recently.

Will fix.

File: test/jdk/jdk/jfr/jcmd/TestJcmdPreserveRepository.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2015, 2018, Oracle and/or its affiliates. All rights reserved.
```
Comment (dholmes-ora) [inline test/jdk/jdk/jfr/jcmd/TestJcmdPreserveRepository.java:2]: New file should have new copyright - unless substantially copied (IIUC).

**Review by dholmes-ora — APPROVED** [review]
Seems okay to me. Can't comment on the actual functionality.

Thanks.

File: src/java.base/share/man/java.1 [hunk 2]
```diff
@@ -1337,6 +1337,12 @@ buffers.
 Maximum number of old objects to track.
 By default, the number of objects is set to 256.
 .TP
+\f[V]preserve-repository=\f[R]{\f[V]true\f[R]|\f[V]false\f[R]}
+Specifies whether files stored in the disk repository should be kept
+after the JVM has exited.
+If false, files are deleted.
+By default, this parameters is disabled.
```
[the remark this answers is not in the document]
Comment (dholmes-ora) [reply src/java.base/share/man/java.1]: Seems an awkward mix of terminology to me.

File: src/jdk.jcmd/share/man/jcmd.1 [hunk 3]
```diff
@@ -402,6 +402,12 @@ On Linux operating systems, the temporary directory is \f[V]/tmp\f[R].
 On Windwows, the temporary directory is specified by the \f[V]TMP\f[R]
 environment variable.)
 .IP \[bu] 2
+\f[V]preserve-repository=\f[R]{\f[V]true\f[R]|\f[V]false\f[R]} :
+Specifies whether files stored in the disk repository should be kept
+after the JVM has exited.
+If false, files are deleted.
+By default, this parameters is disabled.
```
Comment (dholmes-ora) [inline src/jdk.jcmd/share/man/jcmd.1:409]: This still has "parameters" typo.

**Review by mgronlun — APPROVED** [review]
Looks good.

**Review by dholmes-ora — APPROVED** [review]

**Pull request closed without merging**